src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -387,9 +448,9 @@ static void uart_driver_do_write(Context *ctx, GenMessage gen_message)
     free(buffer);
 
     int local_pid = term_to_local_process_id(pid);
-    if (UNLIKELY(memory_ensure_free_with_roots(ctx, TUPLE_SIZE(2), 1, &ref, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
+    if (UNLIKELY(memory_ensure_free_with_roots(ctx, TUPLE_SIZE(2) + REF_SIZE, 1, &ref, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
```

Comment:
You don't need `REF_SIZE` here as `ref` is already allocated.

Reply:
Indeed. Once the hammer was in my hand I started to see too many nails. lol.